### PR TITLE
Basic Auth for providers playlists, sorting and labeling, flussonic catchup (first steps) 

### PR DESCRIPTION
**Pull request by prog1407**
Basic Auth for providers playlists: allows specifying username and password per connection. It is used for basic auth. Mainly needed to support playlists from tvheadend

Advanced sorting: allows selecting on which level to sort channels. Possible options: provider level (channels are sorted on provider level, in the final playlist we get sorted channels of the first provider and then the same for a second), proxy level (all channels are sorted alphabetically in the final joined playlist). It is also possible to specify the channel prefix that will be used in the channel's name. It doesn't affect sorting anyhow and play an indication role for end-user to see which channel comes from which provider.

Flussonic catchup -- first steps. Only catchup of type shift was supported previously. Now flussonic is correctly passed to the end-user's player. However iptv-proxy is not able to handle it properly yet.

### CODE REVIEW

File: src/main/java/com/kvaster/iptv/m3u/M3uParser.java [hunk 1]
```diff
@@ -96,8 +96,14 @@ public static M3uDoc parse(String content) {
                 props.put(m.group(1), m.group(2));
                 line = m.group(3).strip();
             } else {
-                LOG.warn("malformed properties: {}", line);
-                break;
+                if (line.contains(" ")) {
```
Comment (kvaster) [inline src/main/java/com/kvaster/iptv/m3u/M3uParser.java]: Telling the truth, I don't completely understand the purpose of theese changes.

**Review by kvaster — COMMENTED** [review]
I think that this PR should be splitted into several - each featre separately.

**Comment by kvaster** [comment]
Hi. Just to let know I'm not ignoring PRs - I'm a little bit busy right now and will get back in a few days to this PR.

**Comment by prog1407** [comment]
Hello! No worries, I think we can mark this one as obsolete since tvheadend support is already implemented and other features seem to be not that actual for other users. Or could be done by separate PRs.

**Pull request closed without merging**